MobileCenter/MobileCenter/MSMobileCenter.m

```diff
@@ -428,20 +419,44 @@ + (void)resetSharedInstance {
 
 #pragma mark - Application life cycle
 
-#if !TARGET_OS_OSX
 /**
  *  The application will go to the foreground.
  */
 - (void)applicationWillEnterForeground {
+
+  /**
+   * Triggering a resume here to make sure our pipeline is working. While it won't be suspended when going into the
+   * background anymore, it might be suspended because we're offline or it failed to send events.
+   */
   [self.logManager resume];
 }
 
-/**
- *  The application will go to the background.
- */
-- (void)applicationDidEnterBackground {
-  [self.logManager suspend];
+- (void)dealloc {
+  [self removeObservers];
 }
+
+#pragma mark - Observers
+
+- (void)addObservers {
+#if TARGET_OS_OSX
+  [MS_NOTIFICATION_CENTER addObserver:self
```

Comment:
Why are you bring this back? It was removed since we decide not to suspend services on macOS apps.

Reply:
I cleaned this up.